lib/content-services/src/lib/directives/node-counter.directive.ts

```diff
@@ -0,0 +1,55 @@
+/*!
+ * @license
+ * Copyright 2019 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Directive, Input, Component, OnInit, OnChanges, ComponentFactoryResolver, ViewContainerRef } from '@angular/core';
+
+@Directive({
+    selector: '[adf-node-counter]'
+})
+export class NodeCounterDirective implements OnInit, OnChanges {
+    @Input('adf-node-counter')
+    counter: number;
+
+    componentRef: NodeCounterComponent;
+
+    constructor(
+        private resolver: ComponentFactoryResolver,
+        public viewContainerRef: ViewContainerRef
+    ) {}
+
+    ngOnInit() {
+        const componentFactory = this.resolver.resolveComponentFactory(NodeCounterComponent);
```

Comment:
why do you need dynamically creating same component?

Reply:
It is inside the directive that we create the NodeCounterComponent. Not sure if I answered your question or not 😅